v2/AdminCore.Common/Interfaces/IClientService.cs

```diff
@@ -0,0 +1,22 @@
+using AdminCore.DTOs.Client;
+using System.Collections.Generic;
+
+namespace AdminCore.Common.Interfaces
+{
+  public interface IClientService
+  {
+    IList<ClientDto> GetAll();
+
+    void UpdateClient(ClientDto clientDto);
+
+    void CreateClient(ClientDto clientDto);
+
+    void SaveClient(ClientDto clientDto);
+
+    void DeleteClient(int clientId);
+
+    ClientDto GetClientById(int clientId);
+
+    ClientDto GetClientByClientNameContaining(string clientName);
```

Comment:
Not sure about the naming here, "containing". Should be fine just to say GetClientByClientName